docs/DafnyRef/UserGuide.md

```diff
@@ -656,6 +656,84 @@ There are many great options that control various aspects of verifying dafny pro
+
+The mechanism for early detection focuses on measuring the resources
+used to complete a proof (either using duration or a more deterministic
+"resource count" metric available from Z3). Dafny can re-run a given
+proof attempt multiple times after automatically making minor changes to
+the structure of the input or to the random choices made by the solver.
+If the resources used during these attempts (or the ability to find a
+proof at all) vary widely, we say that the verification of the relevant
+properties is _unstable_.
+
+#### 25.7.5.1. Measuring stability
+
+To measure the stability of your proofs, start by using the
+`-randomSeedIterations:N` flag to instruct Dafny to attempt each proof
+goal `N` times, using a different random seed each time. The random seed
+used for each attempt is derived from the global random seed `S`
+specified with `-randomSeed:S`, which defaults to `0`.
+
+For most use cases, it also makes sense to specify the
+`-verificationLogger:csv` flag, to log verification cost statistics to a
+CSV file. By default, the resulting CSV files will be created in the
+`TestResults` folder of the current directory.
+
+Once Dafny has completed, the
+[`dafny-reportgenerator`](https://github.com/dafny-lang/dafny-reportgenerator/)
+tool is a convenient way to process the output. It allows you to specify
+several limits on statistics computed from the elapsed time or solver
+resource use of each proof goal, returning an error code when it detects
+violations of these limits. You can find documentation on the full set
+of options for `dafny-reportgenerator` in its
+[`README.md`](https://github.com/dafny-lang/dafny-reportgenerator/blob/main/README.md)
+file.
+
+In general, we recommend something like the following:
+
+```
+dafny-reportgenerator --max-resource-cv-pct 10 TestResults/*.csv
+```
+
+This bounds the [coefficient of
+variation](https://en.wikipedia.org/wiki/Coefficient_of_variation) of
+the solver resource count at 10% (0.10). We recommend a limit of less
+than 20%, perhaps even as low as 5%. However, when beginning to analyze
+a new project, it may be necessary to set limits as high as a few
+hundred percent and incrementally ratchet down the limit over time.
+
+When first analyzing stability, you may also find that certain proof
+goals succeed on some iterations and fail on others. If your aim is
+first to ensure that instability doesn't worsen and then to start
+improving it, integrating `dafny-reportgenerator` into CI and using the
+`--allow-different-outcomes` flag may be appropriate. Then, once you've
+improved stability sufficiently, you can likely remove that flag (and
+likely have significantly lower limits on other stability metrics).
+
+#### 25.7.5.2. Improving stability
+
+Improving stability is typically closely related to improving performance
+overall. As such, techniques for debugging slow verification
+[^sec-verification-debugging-slow] are typically useful for debugging
```

Comment:
Should this be [#sec-

Reply:
Ah, good point. I changed it from a footnote to a link, too.